internal/observability/config.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package observability sets up and configures observability tools.
+package observability
+
+// ExporterType represents a type of observability exporter.
+type ExporterType string
+
+const (
+	ExporterStackdriver ExporterType = "STACKDRIVER"
+	ExporterPrometheus  ExporterType = "PROMETHEUS"
+	ExporterOCAgent     ExporterType = "OCAGENT"
+	ExporterNoop        ExporterType = "NOOP"
+)
+
+// Config holds all of the configuration options for the observability exporter
+type Config struct {
+	ExporterType ExporterType `env:"OBSERVABILITY_EXPORTER, default=STACKDRIVER"`
+
+	*OpenCensusConfig
+	*StackdriverConfig
+	*OCAgentConfig
+}
+
+// OpenCensusConfig holds the configuration options for the open census exporter
+type OpenCensusConfig struct {
+	TraceProbabilitySampleRate float64 `env:"TRACE_PROBABILITY, default=0.40"`
+}
+
+// StackdriverConfig holds the configuration options for the stackdriver exporter
+type StackdriverConfig struct {
+	ProjectID string `env:"PROJECT_ID"`
+}
+
+// OCAgentConfig holds the configuration options for the default opencensus exporter
+type OCAgentConfig struct {
+	Insecure bool   `env:"OCAGENT_INSECURE, default=true"`
```

Comment:
Should the default be false?

Reply:
Great question. I didnt want to change the default that was there previously, which is why I stuck with true.

Reply:
Let's leave it true, but can you file an issue with the kind/security tag and we'll take a look at changing it.

Reply:
Will do. Just going to wait for a day until either this PR merges or not so that I can link to the correct spot in the code in the issue.